bugwarrior/services/todoist.py

```diff
@@ -0,0 +1,220 @@
+            for section in sections:
+                all_sections.append(section)
+        return all_sections
+
+    def get_users(self, project_id):
+        all_users = []
+        users_iter = self._api.get_collaborators(project_id)
+        for users in users_iter:
+            for user in users:
+                all_users.append(user)
+        return all_users
+
+    def get_issues(self):
+        tasks_iter = self._api.filter_tasks(query=self.filter)
+        return tasks_iter
+
+
+class TodoistIssue(Issue):
+    ASSIGNEE = "todoistassignee"
+    ASSIGNER = "todoistassigner"
+    CONTENT = "todoistcontent"
+    DESCRIPTION = "todoistdescription"
+    DURATION = "todoistduration"
+    ID = "todoistid"
+    SECTION = "todoistsection"
+    URL = "todoisturl"
+
+    PRIORITY_MAP = {
+        4: "H",
+        3: "M",
+        2: "L",
+        1: None,
+    }
+
+    UDAS = {
+        ID: {
+            "type": "string",
+            "label": "Todoist ID",
+        },
+        CONTENT: {
+            "type": "string",
+            "label": "Todoist Content",
+        },
+        DESCRIPTION: {
+            "type": "string",
+            "label": "Todoist Description",
+        },
+        DURATION: {
+            "type": "string",
+            "label": "Todoist Duration",
+        },
+        SECTION: {
+            "type": "string",
+            "label": "Todoist Section",
+        },
+        ASSIGNEE: {
+            "type": "string",
+            "label": "Todoist Assignee",
+        },
+        ASSIGNER: {
+            "type": "string",
+            "label": "Todoist Assigner",
+        },
+        URL: {
+            "type": "string",
+            "label": "Todoist URL",
+        },
+    }
+
+    UNIQUE_KEY = (ID, ID)
+
+    # replace characters that cause escaping issues in teh description like [] and "
+    # this is a workaround for https://github.com/ralphbean/taskw/issues/172
+    def _unescape_content(self, content):
+        return (
+            content.replace('"', "'")  # prevent &dquote; in task details
+            .replace("[", self.config.char_open_bracket)  # prevent &open; and &close;
+            .replace("]", self.config.char_close_bracket)
+        )
+
+    def to_taskwarrior(self):
+        default_time = time(0, 0, 0)
+        # use due date "scheduled".
+        # adjust timezone to use local time for "floating" dates
+        if self.record.due and type(self.record.due.date) is datetime:
+            if self.record.due.timezone:
+                scheduled = self.record.due.date
+            else:
+                scheduled = self.record.due.date.replace(tzinfo=None)
+        else:
+            scheduled = (
+                datetime.combine(self.record.due.date, default_time, tzinfo=None)
+                if self.record.due
+                else None
+            )
+
+        # use deadline as "due".
+        # deadline if set is only a date with no time or timezone. adjust to locla time
```

Comment:
This eliminates a typo but also it seems to be an error -- it isn't adjusted to local time it's just set to midnight with no timezone, right?
```suggestion
        # deadline if set is only a date with no time or timezone.
```

Reply:
> it isn't adjusted to local time it's just set to midnight with no timezone, right?

Correct